milestones/api.py

```diff
@@ -0,0 +1,325 @@
+"""
+api.py is an interface module for Python-level integration with the
+Milestones app.
+
+In the current incarnation of this particular application, the API
+operations are equivalent to the orchestration layer, which manages
+the application's workflows.
+
+Helplful Hint:  When modeling Milestones, I've found that it's helpful
+to first consider the process for fulfilling the Milestone, and then
+consider the process for requiring the Milestone.
+
+Note the terminology difference at this layer vs. Data -- add/edit/get/remove
```

Comment:
I wanted to make it clear which layer was being worked with -- I've found it easier during development to be able to read the method and know that I'm calling the api layer or the data layer -- it also could help us in the future if we want to programmatically/automatically enforce certain access patterns (eg, all data.py modules present fetch(), and only api.py can call fetch())